extensions/azure/vault/src/main/java/org/eclipse/dataspaceconnector/security/azure/AzureVaultExtension.java

```diff
@@ -66,17 +66,20 @@ public void intializeVault(ServiceExtensionContext context) {
             throw new AzureVaultException("'edc.vault.tenantid' must be supplied but was null!");
         }
 
+        String clientSecret = context.getSetting("edc.vault.secret", null);
         String certPath = context.getSetting("edc.vault.certificate", null);
-        if (isNullOrEmpty(certPath)) {
-            throw new AzureVaultException("'edc.vault.certificate' must be supplied but was null!");
+        if (isNullOrEmpty(certPath) && isNullOrEmpty(clientSecret)) {
+            throw new AzureVaultException("Either 'edc.vault.certificate' or 'edc.vault.secret' must be supplied but both was null!");
```

Comment:
Small typo - was--> were

Also, while you are changing that, maybe remove the exclamation mark since we don't want the EDC to yell at users? :-)

Reply:
Corrected!
thanks